server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -126,16 +128,16 @@ public PartitionFilesSorter(MemoryTracker memoryTracker, RssConf conf, AbstractS
               try {
                 while (!shutdown) {
                   FileSorter task = shuffleSortTaskDeque.take();
-                  memoryTracker.reserveSortMemory(reserveMemoryForSingleSort);
+                  this.memoryTracker.reserveSortMemory(initialReserveMemoryForSingleSort);
```

Comment:
I move below both 2 parts  
```
this.memoryTracker.reserveSortMemory(initialReserveMemoryForSingleSort);
                  while (!memoryTracker.sortMemoryReady()) {
                    Thread.sleep(20);
                  }
```
and 
```
 source.startTimer(WorkerSource.SortTime(), task.fileId);
```
into sort to ensure that
1. the result of sortMemoryReady() is correct.
2. the timer metrics is accurate.
But the sortMemoryReady() may need to add a synchronized  to keep thread-safe because it is moved from single thread scenario to multi-thread scenario,  I'm not quite sure will it affect the performance if we add synchronized for this method because I find also add synchronized for releaseSortMemory.